admin/services/data-access/sql.role-connection.builder.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const roles = require('../../lib/consts/roles')
+const sqlConfig = require('../../config/sql.config')
+const config = require('../../config')
+const R = require('ramda')
+
+const builder = {
+  /**
+   * @description builds a sql connection config object
+   * @param {string} roleName - the role of the user to create a connection config for
+   * @param {boolean} readonly - set to true for a readonly connection, typically to a sql azure replica
+   * @returns {object} a config object that works with mssql
+   */
+  build: function build (roleName, readonly = false) {
+    let cfg
+    switch (roleName) {
+      case roles.teacher:
+        cfg = sqlConfig
+        break
+      case roles.techSupport:
+        cfg = R.clone(sqlConfig)
```

Comment:
This is clear code, but as you are using Ramda anyway we could do the clone and the assignment in a single call, and then freeze the result, perhaps something like this:

```
const cfg = Object.freeze(R.mergeDeepRight(sqlConfig, {
  user: 'foo',
  password: 'bar',
  pool: {
    min: 0,
    max: 100
  }
}))
```

https://ramdajs.com/docs/#mergeDeepRight